operator/e2e/tests/cert_manager_test.go

```diff
@@ -0,0 +1,247 @@
+// /*
+// Copyright 2025 The Grove Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+//go:build e2e
+
+package tests
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ai-dynamo/grove/operator/e2e"
+	"github.com/ai-dynamo/grove/operator/e2e/setup"
+	"github.com/ai-dynamo/grove/operator/e2e/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	certManagerIssuerYAML = `
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-issuer
+spec:
+  selfSigned: {}
+`
+	certManagerCertificateYAML = `
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: grove-webhook-server-cert
+  namespace: grove-system
+spec:
+  secretName: grove-webhook-server-cert
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+  dnsNames:
+  - grove-operator
+  - grove-operator.grove-system
+  - grove-operator.grove-system.svc
+  - grove-operator.grove-system.svc.cluster.local
+`
+)
+
+func TestCertManagerIntegration(t *testing.T) {
+	// 1. Prepare cluster
+	ctx := context.Background()
+	clientset, restConfig, dynamicClient, cleanup := prepareTestCluster(ctx, t, 1)
+	defer cleanup()
+
+	// 2. Install cert-manager
+	deps, err := e2e.GetDependencies()
+	if err != nil {
+		t.Fatalf("Failed to get dependencies: %v", err)
+	}
+
+	cmConfig := &setup.HelmInstallConfig{
+		RestConfig:      restConfig,
+		ReleaseName:     deps.HelmCharts.CertManager.ReleaseName,
+		ChartRef:        deps.HelmCharts.CertManager.ChartRef,
+		ChartVersion:    deps.HelmCharts.CertManager.Version,
+		Namespace:       deps.HelmCharts.CertManager.Namespace,
+		CreateNamespace: true,
+		Wait:            true,
+		Timeout:         10 * time.Minute, // cert-manager needs time to download images and start
+		RepoURL:         deps.HelmCharts.CertManager.RepoURL,
+		Values: map[string]interface{}{
+			"installCRDs": true,
+		},
+		HelmLoggerFunc: logger.Debugf,
+		Logger:         logger,
+	}
+
+	logger.Info("Installing cert-manager...")
+	if _, err := setup.InstallHelmChart(cmConfig); err != nil {
+		t.Fatalf("Failed to install cert-manager: %v", err)
+	}
+
+	// Wait for cert-manager webhook to be ready (InstallHelmChart with Wait=true should handle it, but to be safe)
+	if err := utils.WaitForPodsInNamespace(ctx, cmConfig.Namespace, restConfig, 3, 2*time.Minute, 5*time.Second, logger); err != nil {
```

Comment:
Do we require 2 minutes? Can we try defaultPullTimeout instead?

Also just use defaultPullInterval here (it's also 5 seconds).